routers/web/repo/migrate.go

```diff
@@ -257,3 +258,19 @@ func setMigrationContextData(ctx *context.Context, serviceType structs.GitServic
 	ctx.Data["Services"] = append([]structs.GitServiceType{structs.PlainGitService}, structs.SupportedFullGitService...)
 	ctx.Data["service"] = serviceType
 }
+
+func MigrateCancelPost(ctx *context.Context) {
+	migratingTask, err := admin_model.GetMigratingTask(ctx.Repo.Repository.ID)
+	if err != nil {
```

Comment:
`if errors.Is(err, util.ErrNotExist) {…} else if err != nil {`?

Reply:
No difference, this handler is only for the special JS. JS only accepts 200, and retries for others.

Reply:
Given that I've already seen PRs doing nothing else than handling a `404` not as a `500`, I'd disagree.

Reply:
No 404 or 500 any more, just redirect. 8f38e57ac5ce7761e3f52b5f1f6e238b8f29d76b